### PR TITLE
Fix #30 ssh host key files empty

### LINKED ISSUE #30: SSH host key files empty
Due to some race-condition or possibly no space(before expanding the filesystem), the /etc/ssh/ssh_host* files are empty after first boot of some Mirtes.

No known real fix, but add patch that checks those files and if empty regenerate them and restart ssh

### PR DESCRIPTION
**Pull request by ArendJan**
Fix #30 by checking the ssh_host files at every boot and regenerating them when one of them is empty.

### CODE REVIEW

File: network_setup.sh [hunk 1]
```diff
@@ -100,6 +100,30 @@ function check_connection {
 	fi
 }
 
+function check_ssh_host_keys() {
+	# Sometimes these files are empty after first boot and ssh won't work.
```
Comment (mklomp) [inline network_setup.sh:104]: Could you add the links to github issues about others that ran intro similar issues.

File: network_setup.sh [hunk 2]
```diff
@@ -100,6 +100,30 @@ function check_connection {
...
+	# This will check the files for size and regenerate them when one of them is empty
+	if
+		file_empty "/etc/ssh/ssh_host_ecdsa_key" || file_empty "/etc/ssh/ssh_host_ecdsa_key.pub" ||
+			file_empty "/etc/ssh/ssh_host_ed25519_key" || file_empty "/etc/ssh/ssh_host_ed25519_key.pub" ||
+			file_empty "/etc/ssh/ssh_host_rsa_key" || file_empty "/etc/ssh/ssh_host_rsa_key.pub"
+	then
+		echo "Regenerating ssh host keys and restarting sshd"
+		ssh-keygen -A
```
Comment (mklomp) [inline network_setup.sh:112]: Also for the other commands: do we need sudo? Not sure how this script will be run....

**Review by mklomp — APPROVED** [review]
Nice fix!

File: network_setup.sh (hunk 2 above)
Comment (ArendJan) [reply network_setup.sh:112]: The script is started by systemctl with privileges, so sudo is not required. Also quite a lot of other commands(nmcli) in that script need privileges when running, but are started without sudo.

File: network_setup.sh (hunk 1 above)
Comment (ArendJan) [reply network_setup.sh:104]: Added!

File: network_setup.sh [hunk 3]
```diff
@@ -109,7 +134,7 @@ $MIRTE_SRC_DIR/mirte-install-scripts/usb_ethernet.sh
 # be generated on first boot (so not when generating
 # the image in network_setup.sh)
 if [ ! -f /etc/ssid ] || [[ $(cat /etc/hostname) == "Mirte-XXXXXX" ]]; then
-	UNIQUE_ID=$(openssl rand -hex 3)
+	UNIQUE_ID=$(tr -cd "1-9A-F" </dev/urandom | head -c 6)
```
Comment (mklomp) [inline network_setup.sh:137]: This one should be removed from this PR

Comment (ArendJan) [reply network_setup.sh]: Hmm, some strange things with git, now that change is gone

**Pull request merged**